### PR TITLE
Increase reconcile timeout

### PR DESCRIPTION
**Pull request by yhaliaw**
### Overview

Increase reconcile timeout length.

### Rationale

Reconcile event timeout can be longer than the default which was half of the event interval.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated.
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -499,7 +499,9 @@ def _on_config_changed(self, _event: ConfigChangedEvent) -> None:
         self._refresh_firewall()
         try:
             self._event_timer.ensure_event_timer(
-                "reconcile-runners", self.config["reconcile-interval"]
+                event_name="reconcile-runners",
+                interval=int(self.config["reconcile-interval"]),
```
Comment (yhaliaw) [inline src/charm.py:503]: The conversion to int and validation of the `reconicle-interval` configuration, will be done in a future PR where all configuration in the `charm.py` is refactor to `charm_state.py`.

File: src/charm.py [hunk 2]
```diff
@@ -499,7 +499,9 @@ def _on_config_changed(self, _event: ConfigChangedEvent) -> None:
...
+                timeout=int(self.config["reconcile-interval"]) - 1,
```
Comment (cbartz) [inline src/charm.py:504]: Is it really safe to assume that we do not need a pause of more than one second between two reconciliations?

**Review by cbartz — DISMISSED** [review]
Left a question.

File: src/charm.py (hunk 2 above)
Comment (yhaliaw) [reply src/charm.py:504]: The unit is 1 minutes, which I assume is enough.

Comment (cbartz) [reply src/charm.py:504]: For interval, its minutes, for timeout, its actual seconds. 

```
man timeout

DURATION is a floating point number with an optional suffix: 's' for seconds (the default), 'm' for minutes, 'h' for hours or 'd' for days.  A duration of 0 disables the associated timeout
```

Comment (yhaliaw) [reply src/charm.py:504]: I update the code to be more consistent.
The `interval` was in minutes, but the `timeout` was in seconds.
Now both are in minutes.

**Review by jdkandersson — APPROVED** [review]

**Review by yanksyoon — APPROVED** [review]

File: src/event_timer.py [hunk 3]
```diff
@@ -69,15 +69,21 @@ def ensure_event_timer(self, event_name: str, interval: int, timeout: Optional[i
...
 
         Raises:
             TimerEnableError: Timer cannot be started. Events will be not emitted.
         """
+        if timeout is not None:
+            timeout_in_secs = timeout * 60
+        else:
+            timeout_in_secs = interval * 30
+
         context: EventConfig = {
```
Comment (yanksyoon) [inline src/event_timer.py:82]: A nitpicky comment, the `EventConfig` definition could have attributes docstring so the time units can be determined :)

Comment (yanksyoon) [reply src/event_timer.py:82]: Pleaes do feel free to resolve this for now!

**Review by cbartz — APPROVED** [review]

File: src/event_timer.py (hunk 3 above)
Comment (yhaliaw) [reply src/event_timer.py:82]: I have added the attribute docstring in the refactoring charm_state branch.

**Pull request merged**